week-3/Homework/mandatory/5-journey-planner.js

```diff
@@ -1,14 +1,129 @@
+ Hint: search for string methods like Includes and IndexOf.
+*/
+
+function checkCodeIsThere(stringText) {
+  let magicWord = "code";
+  //edit code below
+  if (stringText.includes("code")) {
+    return stringText.indexOf("code");
+  } else {
+    return "Not found";
+  }
+}
+
 /*
   I am new to London and would like to know what transport I can take to different famous locations.
-  An array with London locations have been provided.
+  The input provided contains a list of locations in London. Each of locations is followed by a list
+  of transport modes that can be used to get there.
+  Let's see an example:
+  
+  To take to Tower Bridge, you can use tube or river boat. This information will represented as 
+    ["Tower Bridge", "tube", "river boat"]
+  Where
+    the 1st element says the name of the location,
+    and rest of them says the transport modes.
+  You will then get a list of these information, e.g:
+  [
+    ["Tower Bridge", "tube", "river boat"],
+    ["Abbey road", "double decker"],
+    ["London Eye", "tube", "river boat", "bus"]
+  ]
+  You have to finish up the body of journeyPlanner function that should tell where I can go if I only
+  want to use a specific mode of transport. But before jumping straight to the main function, we will
+  break down the whole task into smaller steps that make our job easier.
+  This technic is also referred as problem decomposition. It helps you to reduce scope of the problem 
+  by only focusing on a small chunk of the whole problem at a time.)
+*/
+
+/*
+  Implement the function getTransportModes that
+   - Accepts an array containing the location and available transport modes
+     e.g: ["Tower Bridge", "tube", "river boat"]
+   - Returns an array including the available transport modes to the given location
+     e.g: ["tube", "river boat"]
+  
+  Hint: Use the corresponding array method to split the array.
+*/
+function getTransportModes(mode) {
+  // let transportMode = mode.slice(1);
+  // return transportMode;
 
-  Return an array of where I can go if I only want to use a specific mode of transport.
+  return mode.slice(1); // returns last 2 items of the array, does not have the first item
+}
 
-  NOTE: only the names should be returned, not the means of transport.
+/*
+  Implement the function isAccessibleByTransportMode that
+   - Accepts two parameters:
+     1) First parameter is an array of transport modes
+        e.g: ["tube", "river boat"]
+     2) Second parameter is a string containing a transport mode
+        e.g: "river boat"
+     
+   - Returns 
+     * True if the location in the first parameter is accessible by the transport mode given in second parameter
+     * Otherwise, returns false
+  Hint: Use the corresponding array method to decide if an element is member of an array.
 */
+function isAccessibleByTransportMode(array1, array2) {
+  if (array1.includes(array2)) {
+    return true;
+  } else {
+    return false;
+  }
+}
```

Comment:
The includes() method returns true or false, so you could simply do this: 
`return array1.includes(array2)`

Again, array names could be more meaningful here 🙂

Reply:
As @dsobkow outlined, the expression in the if statement results in a Boolean value, in other words, it will be either `true` or `false` thereby you don't have to manually say that this case means `true` and the other is `false`.

Reply:
Basicelly method `includes` returns either `true` or `false` according to the condition.